Model/Subscription.php

```diff
@@ -49,4 +49,9 @@ public function setProductOption(array $productOptions)
         $platformSpecificFields[self::PLATFORM_FIELD_KEY][self::PRODUCT_OPTION] = $productOptions;
         return $this->setPlatformSpecificFields($platformSpecificFields);
     }
+
+    public function getUserDefinedFields()
+    {
+       return json_encode(parent::getUserDefinedFields());
```

Comment:
Notes after review: Need to research return types to see if we can get the interface to return the userDefinedFields as an object instead of an array, instead of using json_encode in the model. Other areas that request the userDefinedFields may not expect a json string.